tests/cross_functional/system_test/test_cluster_wide_key_rotation_systemtest.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import pytest
+
+from ocs_ci.framework.pytest_customization.marks import (
+    system_test,
+    magenta_squad,
+    ignore_leftovers,
+)
+from ocs_ci.framework.testlib import E2ETest
+from ocs_ci.helpers.e2e_helpers import (
+    Run_fio_till_cluster_full,
+)
+from ocs_ci.helpers.keyrotation_helper import (
+    verify_key_rotation_time,
+    enable_key_rotation,
+    verify_new_key_after_rotation,
+    OSDKeyrotation,
+)
+from ocs_ci.helpers.osd_resize import basic_resize_osd, get_storage_size
+from ocs_ci.helpers.sanity_helpers import Sanity
+from ocs_ci.ocs import warp
+
+log = logging.getLogger(__name__)
+
+
+@magenta_squad
+@system_test
+@ignore_leftovers
+class TestKeyRotationWithClusterFull(E2ETest):
+    """
```

Comment:
I would suggest to move this doc string under test method instead of test class. Because this test class get more test methods in future.